hystrix-core/src/main/java/com/netflix/hystrix/collapser/CollapsedRequestObservableFunction.java

```diff
@@ -37,14 +44,23 @@
  */
 /* package */class CollapsedRequestObservableFunction<T, R> implements CollapsedRequest<T, R>, OnSubscribe<T> {
     private final R argument;
-    private final AtomicReference<CollapsedRequestObservableFunction.ResponseHolder<T>> rh = new AtomicReference<ResponseHolder<T>>(new CollapsedRequestObservableFunction.ResponseHolder<T>());
-    private final BooleanSubscription subscription = new BooleanSubscription();
+    private final Subject<T, T> responseSubject = ReplaySubject.create();
 
     public CollapsedRequestObservableFunction(R arg) {
         this.argument = arg;
     }
 
     /**
+     * This is a passthrough that allows a Subscriber to receive values from the Subject that collapsers are writing values/errors into
+     * The one interesting bit is that they may do so in a completely unbounded way.  There's no way to express
+     * backpressure currently that makes sense other than to buffer this stream and allow it to grow unboundedly
+     */
+    @Override
+    public void call(Subscriber<? super T> observer) {
+        responseSubject.onBackpressureBuffer().subscribe(observer);
```

Comment:
I think you can also use `unsafeSubscribe` to avoid an allocation, as I don't think you need the `unsubscribe` behavior on termination, nor do you need error handling, as long as you control the `Subscriber` used here and don't do anything wrong in it.